src/Data/ByteString/Conversion/From.hs

```diff
@@ -54,10 +54,11 @@ runParser p b = case feed (parse p b) "" of
 
 runParser' :: Parser a -> Lazy.ByteString -> Either String a
 runParser' p b = case Lazy.parse p b of
-    Lazy.Done ""   r -> Right r
-    Lazy.Done _    _ -> Left "Trailing input"
-    Lazy.Fail _ [] m -> Left m
-    Lazy.Fail _ x  m -> Left (shows x . showString m $ "")
+    Lazy.Done t    r
+        | Lazy.null t -> Right r
```

Comment:
The IsString instance is not present with old enough dependencies, unfortunately

Sent from my iPhone

> On 23 Jan 2016, at 00:44, Toralf Wittner <notifications@github.com> wrote:
> 
> In src/Data/ByteString/Conversion/From.hs:
> 
> > @@ -54,10 +54,11 @@ runParser p b = case feed (parse p b) "" of
> >  
> >  runParser' :: Parser a -> Lazy.ByteString -> Either String a
> >  runParser' p b = case Lazy.parse p b of
> > -    Lazy.Done ""   r -> Right r
> > -    Lazy.Done _    _ -> Left "Trailing input"
> > -    Lazy.Fail _ [] m -> Left m
> > -    Lazy.Fail _ x  m -> Left (shows x . showString m $ "")
> > +    Lazy.Done t    r
> > +        | Lazy.null t -> Right r
> The String literal pattern does not work?
> 
> —
> Reply to this email directly or view it on GitHub.
>